pkg/platform/runtime/buildexpression/buildexpression.go

```diff
@@ -906,6 +906,31 @@ func (e *BuildExpression) SetDefaultTimestamp() error {
 	return nil
 }
 
+// MaybeSetDefaultTimestamp changes the solve node's "at_time" value to "$at_time" if and only if
+// the current value is the given timestamp.
+// Buildscripts prefer to use variables for at_time and define them outside the buildexpression as
+// the expression's commit time.
+// While modern buildexpressions use variables, older ones bake in the commit time. This function
+// exists primarily to update those older buildexpressions for use in buildscripts.
+func (e *BuildExpression) MaybeSetDefaultTimestamp(ts *strfmt.DateTime) error {
+	if ts == nil {
+		return nil // nothing to compare to
+	}
+	atTimeNode, err := e.getSolveAtTimeValue()
+	if err != nil {
+		return errs.Wrap(err, "Could not get %s node", AtTimeKey)
+	}
+	if strings.HasPrefix(*atTimeNode.Str, "$") {
+		return nil
+	}
```

Comment:
Why are we returning in this case? Can we add a comment to explain why.

Reply:
This is covered in the doc comment for the function:

```
MaybeSetDefaultTimestamp changes the solve node's "at_time" value to "$at_time" if and only if
the current value is the given timestamp.
```

If the node has a "$" prefix (i.e. it is "$at_time"), then we don't change it. (It's already a default timestamp.)

Reply:
> If the node has a "$" prefix (i.e. it is "$at_time"), then we don't change it. (It's already a default timestamp.)

While this part isn't specifically in the docstring, reading it and seeing this makes way more sense now. Thanks!